jbpm-flow/src/main/java/org/jbpm/workflow/instance/node/RuleSetNodeInstance.java

```diff
@@ -72,32 +77,61 @@ public void internalTrigger(final NodeInstance from, String type) {
     	if ( !org.jbpm.workflow.core.Node.CONNECTION_DEFAULT_TYPE.equals( type ) ) {
             throw new IllegalArgumentException( "A RuleSetNode only accepts default incoming connections!" );
         }
-    	// first set rule flow group
-    	setRuleFlowGroup(resolveRuleFlowGroup(getRuleSetNode().getRuleFlowGroup()));
-    	
-    	//proceed
+    	RuleSetNode ruleSetNode = getRuleSetNode();
     	KnowledgeRuntime kruntime = getProcessInstance().getKnowledgeRuntime();
-    	Map<String, Object> inputs = evaluateParameters(getRuleSetNode());
-    	for (Entry<String, Object> entry : inputs.entrySet()) {
-    	    String inputKey = getRuleFlowGroup() + "_" +getProcessInstance().getId() +"_"+entry.getKey();
-    	    
-    	    factHandles.put(inputKey, kruntime.insert(entry.getValue()));
-    	}
+    	Map<String, Object> inputs = evaluateParameters(ruleSetNode);
     	
-    	if (actAsWaitState()) {
-        	addRuleSetListener();
-            ((InternalAgenda) getProcessInstance().getKnowledgeRuntime().getAgenda())
-            	.activateRuleFlowGroup( getRuleFlowGroup(), getProcessInstance().getId(), getUniqueId() );
-            
+    	if (ruleSetNode.isDMN()) {
+    	    String namespace = resolveVariable(ruleSetNode.getNamespace());
+    	    String model = resolveVariable(ruleSetNode.getModel());
+    	    String decision = resolveVariable(ruleSetNode.getDecision());
+    	    
+    	    DMNRuntime runtime = ((StatefulKnowledgeSession) kruntime).getKieRuntime(DMNRuntime.class);
+    	    DMNModel dmnModel = runtime.getModel(namespace, model);
+    	    if (dmnModel == null) {
+    	        throw new IllegalArgumentException( "DMN model '" + model + "' not found with namespace '" + namespace + "'");
+    	    }
```

Comment:
I am not sure how jbpm handles this kind of errors, but yes, it would be an exception at runtime, similar to returning a null model above, except that it does return a model but with compilation/loading errors (i.e., an invalid model).